src/main/java/hudson/plugins/emailext/EmailExtStep.java

```diff
@@ -0,0 +1,191 @@
+import javax.management.Descriptor;
+
+/**
+ * Created by acearl on 9/14/2015.
+ */
+
+public class EmailExtStep extends AbstractStepImpl {
+
+    public final String subject;
+
+    public final String body;
+
+    @CheckForNull
+    private String attachmentsPattern;
+
+    @CheckForNull
+    private String to;
+
+    @CheckForNull
+    private String replyTo;
+
+    @CheckForNull
+    private String mimeType;
+
+    private boolean attachLog;
+
+    private boolean compressLog;
+
+    @DataBoundConstructor
+    public EmailExtStep(String subject, String body) {
+        this.subject = subject;
+        this.body = body;
+    }
+
+    public @CheckForNull String getAttachmentsPattern() {
+        return attachmentsPattern == null ? "" : attachmentsPattern;
+    }
+
+    @DataBoundSetter
+    public void setAttachmentsPattern(@CheckForNull String attachmentsPattern) {
+        if(StringUtils.isNotBlank(attachmentsPattern)) {
+            this.attachmentsPattern = attachmentsPattern;
+        }
+    }
+
+    public @CheckForNull String getTo() {
+        return to == null ? "" : to;
+    }
+
+    @DataBoundSetter
+    public void setTo(@CheckForNull String to) {
+        this.to = Util.fixNull(to);
+    }
+
+    public @CheckForNull String getReplyTo() {
+        return replyTo == null ? "" : replyTo;
+    }
+
+    @DataBoundSetter
+    public void setReplyTo(@CheckForNull String replyTo) {
+        this.replyTo = Util.fixNull(replyTo);
+    }
+
+    public @CheckForNull String getMimeType() {
+        return mimeType == null ? "" : mimeType;
+    }
+
+    @DataBoundSetter
+    public void setMimeType(@CheckForNull String mimeType) {
+        this.mimeType = Util.fixNull(mimeType);
+    }
+
+    public boolean getAttachLog() {
+        return attachLog;
+    }
+
+    @DataBoundSetter
+    public void setAttachLog(boolean attachLog) {
+        this.attachLog = attachLog;
+    }
+
+    public boolean getCompressLog() {
+        return compressLog;
+    }
+
+    @DataBoundSetter
+    public void setCompressLog(boolean compressLog) {
+        this.compressLog = compressLog;
+    }
+
+    public static class EmailExtStepExecution extends AbstractSynchronousNonBlockingStepExecution<Void> {
+
+        private static final long serialVersionUID = 1L;
+
+        @Inject
+        private transient EmailExtStep step;
+
+        @StepContextParameter
+        private transient TaskListener listener;
+
+        @StepContextParameter
+        private transient Launcher launcher;
+
+        @StepContextParameter
+        private transient Run<?,?> run;
+
+        @StepContextParameter
+        private transient FilePath workspace;
+
+        @Override
+        protected Void run() throws Exception {
+            ExtendedEmailPublisher publisher = new ExtendedEmailPublisher();
+            publisher.configuredTriggers.clear();
+
+            AlwaysTrigger.DescriptorImpl descriptor = Jenkins.getInstance().getDescriptorByType(AlwaysTrigger.DescriptorImpl.class);

```

Comment:
`Jenkins.getActiveInstance()` to avoid NPE (although Jenkins instance must be always available when running a step).